dna/models.py

```diff
@@ -814,6 +815,93 @@ def predict_regression(self, data, *, verbose=False):
 
         return predictions
 
+class LinearRegressionBaseline(RegressionModelBase):
+    def __init__(self, seed=0):
+        RegressionModelBase.__init__(self, seed=seed)
+        self.regressor = linear_model.LinearRegression()
+        self.fitted = False
+        self.pipeline_key = 'pipeline'
+        self.steps_key = 'steps'
+        self.prim_name_key = 'name'
+
+    def fit(self, data, *, validation_data=None, output_dir=None, verbose=False):
+        self.fitted = True
+        self.one_hot_primitives_map = self._one_hot_encode_mapping(data)
+        X_data, y = self.prepare_data_for_regression(data)
+        self.regressor.fit(X_data, y)
+
+    def predict_regression(self, data, *, verbose=False):
+        if self.fitted is False:
+            raise ModelNotFitError('LinearRegressionBaseline not fit')
+            
+        X_data, y = self.prepare_data_for_regression(data) 
+        predictions = self.regressor.predict(X_data)
+        return predictions
+    
+    def prepare_data_for_regression(self, data):
+        full_data = pd.DataFrame(data)
+        y = full_data["test_f1_macro"]
+
+        # this takes a column of lists of ints and expands it out into a dataframe of ints
+        metafeature_df = pd.DataFrame(full_data.metafeatures.values.tolist()).reset_index(drop=True)
+        # metafeature_df = pd.DataFrame(np.nan_to_num(metafeature_df_raw.values))
+        assert np.isnan(metafeature_df.values).sum() == 0, "Was not able to impute the metafeatures: nans exist"
+        assert np.isinf(metafeature_df.values).sum() == 0, "Was not able to impute the metafeatures: infs exist"
+
+        # one hot encode by primitive
+        one_hot_encoding = self.one_hot_encode_pipelines(data)
+        assert np.isnan(one_hot_encoding.values).sum() == 0, "Was not able to impute the primitive encoding: nans exist"
+        assert np.isinf(one_hot_encoding.values).sum() == 0, "Was not able to impute the primitive encoding: infs exist"
+
+        # remove unused columns, one hot encode the structure type
+        needed_data = full_data.drop(["test_f1_macro", "problem_type", "pipeline", "metafeatures", "dataset_id"], axis=1).reset_index(drop=True)
+        needed_data = pd.get_dummies(data=needed_data, columns=['pipeline_structure'])
```

Comment:
After digging into this, I see that `test_time` and `pipeline_structure` are included as predictive variables. `test_time` is not known until the pipeline is fully trained and used to make predictions, so it cannot be used as a predictive variable.

Using `pipeline_structure` gives the regression model more information, but it then cannot make predictions about new pipeline structures. I am inclined to not include it either. Thoughts?

Reply:
Yes, I took out `pipeline_structure` in the next PR because I thought the same thing.  And `test_time` is currently all 0 so it isn't affecting anything, but you're totally rught that we should remove it.  Let me make these changes real quick.

Reply:
I am working on it, so don't change it.

Reply:
Oh perfect, then that's sooner than I'll get to it.